patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -2915,38 +2952,69 @@
  			int num = genRand.Next(4);
  			int num2 = 0;
  			int num3 = 0;
-@@ -50715,7 +_,11 @@
+@@ -50715,6 +_,21 @@
  					if (CountNearBlocksTypes(num11, num12, 2, 1, 27) > 0)
  						continue;
  
++					int preHookType = Main.tile[num11, num12].type;
 +					if (!TileLoader.Convert(num11, num12, 1)) {
 +						if (genRand.Next(2) == 0)
 +							flag2 = true;
```

Comment:
I think this should only be when the type changes. If the conversion just gets blocked, no need to try again elsewhere.